src/main/java/seedu/recipe/model/recipe/PropertyNameContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,75 @@
+package seedu.recipe.model.recipe;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import seedu.recipe.commons.util.StringUtil;
+
+/**
+ * Tests that the property name of a {@code Recipe} matches any of the keywords given.
+ * A property is a field in the Recipe, e.g. its Name, Tags, or Ingredients.
+ *
+ * @param <T> Type of the property to be tested
+ */
+public class PropertyNameContainsKeywordsPredicate<T> implements Predicate<Recipe> {
+    private final List<String> keywords;
+    /**
+     * This method gets the name string of property T from Recipe.
+     */
+    private final Function<Recipe, T> propertyGetter;
+    private final Function<T, String> nameGetter;
+
+
+    /**
+     * Constructs an instance of PropertyNameContainsKeywordsPredicate.
+     *
+     * @param keywords       A list of keywords to test against the property name
+     * @param propertyGetter A method that returns the property from a Recipe
+     * @param nameGetter     A method that gets the name of a property
+     */
+    public PropertyNameContainsKeywordsPredicate(List<String> keywords, Function<Recipe, T> propertyGetter,
+                                                 Function<T, String> nameGetter) {
```

Comment:
Personal preference but line continuation might be better if the second line is at indentation level +4 wrt function body

```java
    methodName(...
            paramX, ... ) {
        //Statement 1 of function
```

Reply:
there doesn't seem to be an option for this under intelliJ code style, so i'm leaving it as-is